javascript/javascript3/week1/weatherApp.js

```diff
@@ -0,0 +1,171 @@
+
+    feelsLike.innerHTML = `<b>Feels : ${Math.floor(data.main.feels_like - 273.15)} °C </b>`
+
+    minMaxTemp.innerHTML = `<b>Max Temp : ${Math.floor(data.main.temp_max - 273.15)} °C - Min Temp: ${Math.floor(data.main.temp_min - 273.15)} °C </b>`
+
+    pressure.innerHTML = `<b>Pressure : ${data.main.pressure} hPa - Humidity: ${data.main.humidity}% </b>`
+
+    img.src = `http://openweathermap.org/img/w/${data.weather[0].icon}.png`
+
+    wind.innerHTML = `Wind Speed : ${data.wind.speed} m/s - Wind Degree : ${data.wind.deg}`
+
+    cloud.innerHTML = `Clouds : ${data.clouds.all} %`
+
+    sunTimes.innerHTML = `Sunrise Time: ${sun(data.sys.sunrise)} - Sunset Time : ${sun(data.sys.sunset)}`
+
+    mapLocation.style.visibility = 'visible'
+    const srcLoc = `https://maps.google.com/maps?q=${data.coord.lat},${data.coord.lon}&z=15&output=embed`
+    mapLocation.src = srcLoc
+}
+
+// Get results with automatically latitude and longitude
+
+function getLatitudeLongitude() {
+    function confirmed(location) {
+        const latitude = location.coords.latitude
+        const longitude = location.coords.longitude
+        fetch(`http://api.openweathermap.org/data/2.5/onecall?lat=${latitude}&lon=${longitude}&appid=1e55d7a6d0d20e41c51f61a54d2abb7d`)
+            .then(res => res.json())
+            .then((data) => {
+                if (data.cod != 400) {
+                    informationwithLocation(data)
+                }
+                else {
+                    error()
+                    console.log(data)
+                }
+            })
+    }
+
+    function error() {
+        city.textContent = 'Unable to retrieve your location'
+    }
+
+    if (!navigator.geolocation) {
+        city.textContent = 'Geolocation is not supported by your browser';
+    }
+    else {
+        city.textContent = 'Locating…';
+        navigator.geolocation.getCurrentPosition(confirmed, error);
+    }
+}
+
+// Html information part with Latitude and Longitude automatically
+
+function informationwithLocation(data) {
+    city.innerHTML = `Timezone : ${data.timezone}`
+
+    temprature.innerHTML = `${Math.floor(data.current.temp - 273.15)} °C </b>`
+
+    feelsLike.innerHTML = `<b>Feels : ${Math.floor(data.current.feels_like - 273.15)} °C </b>`
+
+    pressure.innerHTML = `<b>Pressure : ${data.current.pressure} hPa - Humidity: ${data.current.humidity}% </b>`
+
+    img.src = `http://openweathermap.org/img/w/${data.current.weather[0].icon}.png`
+
+    wind.innerHTML = `Wind Speed : ${data.current.wind_speed} m/s - Wind Degree : ${data.current.wind_deg}`
+
+    cloud.innerHTML = `Clouds : ${data.current.clouds} %`
+
+    sunTimes.innerHTML = `Sunrise Time: ${sun(data.current.sunrise)} - Sunset Time : ${sun(data.current.sunset)}`
+
+    mapLocation.style.visibility = 'visible'
+    const srcLoc = `https://maps.google.com/maps?q=${data.lat},${data.lon}&z=15&output=embed`
+    mapLocation.src = srcLoc
+}
```

Comment:
It looks to me like this function is very similar to the informationWithCity function - would it be possible to make it into a single function? It would save you quite a bit of lines of code that would otherwise have to be maintained 2 places, but it also makes it a little easier for the reader to follow.

Reply:
I am using 2 different apis for that and they objects and informations are different from each other. For that I had to do that with 2 different function. Thanks 👍